chain/chunks/src/test/multi.rs

```diff
@@ -1,124 +1,327 @@
     }
+    test
+}
+
+/// Tests that when we have some block producers (validators) in the network,
+/// and one block producer produces a chunk, the chunk is distributed to the
+/// other block producers properly and all other block producers report the
+/// completion of the chunk to the client.
+#[test]
+fn test_distribute_chunk_basic() {
+    let mut test = basic_setup(BasicSetupConfig {
+        block_producers: (0..10).map(|idx| format!("validator_{}", idx).parse().unwrap()).collect(),
+        chunk_only_producers: Vec::new(),
+        epoch_length: 4, // arbitrary
+        num_shards: 3,   // arbitrary
+        track_all_shards: false,
+    });
+    let chunk_producer = test.data[0].chain.next_chunk_producer(1);
+    let chunk_producer_idx = test.data.index_for_account(&chunk_producer);
+    test.sender().for_index(chunk_producer_idx).send_adhoc_event("produce chunk", |data| {
+        let chunk = data.chain.produce_chunk(1);
+        data.chain.distribute_chunk(&chunk);
+    });
+    // Two network rounds is enough because each node should have
+    // forwarded the parts to those block producers that need them.
+    test.run_for(NETWORK_DELAY * 2);
 
-    // Have the first validator distribute a chunk.
-    let fixture = ChunkTestFixture::default();
-    sender.send((
-        0,
-        ShardsManagerRequestFromClient::DistributeEncodedChunk {
-            partial_chunk: fixture.make_partial_encoded_chunk(&fixture.all_part_ords),
-            encoded_chunk: fixture.mock_encoded_chunk.clone(),
-            merkle_paths: fixture.mock_merkle_paths.clone(),
-            outgoing_receipts: fixture.mock_outgoing_receipts.clone(),
+    // All other nodes should have received the chunk header and their owned
+    // parts, but only those that track the shard should have persisted the
+    // entire chunk.
+    for idx in 0..test.data.len() {
+        if idx == chunk_producer_idx {
+            continue;
+        }
+        let data = test.data.get(idx).unwrap();
+        assert_eq!(data.client_events.len(), 2);
+        match &data.client_events[0] {
+            ShardsManagerResponse::ChunkHeaderReadyForInclusion {
+                chunk_producer: producer,
+                ..
+            } => {
+                assert_eq!(producer, &chunk_producer);
+            }
+            _ => panic!("Unexpected event"),
+        }
+        match &data.client_events[1] {
+            ShardsManagerResponse::ChunkCompleted { partial_chunk, shard_chunk } => {
+                if data.chain.cares_about_shard_this_or_next_epoch(1) {
+                    assert_eq!(
+                        partial_chunk.parts().len(),
+                        data.chain.epoch_manager.num_total_parts()
+                    );
+                    assert!(shard_chunk.is_some());
+                } else {
+                    assert_eq!(partial_chunk.parts().len(), 1);
+                    assert_eq!(partial_chunk.parts()[0].part_ord as usize, idx);
```

Comment:
Is `part_ord` an index of chunk part? It looks suspicious that it perfectly matches order of validators because I expect some randomization of assignment.
Could be nice to have `PartialEncodedChunkPart` documented at some point.

Reply:
It doesn't appear random, and I'm guessing that's because it's not important to randomize the part ords. There's no semantic difference between any of the part ords - it's just some arbitrary numbering.